lib/DDG/Spice/NobelPrize.pm

```diff
@@ -0,0 +1,43 @@
+package DDG::Spice::NobelPrize;
+
+# ABSTRACT: Write an abstract here
+
+# Start at http://docs.duckduckhack.com/walkthroughs/forum-lookup.html if you are new
+# to instant answer development
+
+use DDG::Spice;
+use Data::Dumper;
+
+spice is_cached => 1;
+spice proxy_cache_valid => "200 1d"; # defaults to this automatically
+
+spice wrap_jsonp_callback => 1;
+
+spice to => 'http://api.nobelprize.org/v1/prize.json?' 
+                . '&year=$1' 
+                . '&category=$2';
+
+
+# Triggers - https://duck.co/duckduckhack/spice_triggers
+triggers startend => 'nobel';
+
+spice from => '([^-]+)-([^-]+)';
+
+# Handle statement
+handle query_clean => sub {
```

Comment:
The code below had a problem:
``` perl
    return unless m/$year_regex|$category_regex/;

    my $year = $+{year} || "";
    my $category = $+{category} || "";
```
that only the last match between the two above would remember the named match pattern. So I had to change to:
``` perl
return unless m/$year_regex|$category_regex/;
    
    m/$year_regex/;
    my $year = $+{year} || "";
    
    m/$category_regex/;
    my $category = $+{category} || "";
```

This way we ensure that the key (`year` or `category`) we are looking for is sure to be present with a matched pattern when the pattern exists.